docs/content/posts/2023-03-09-release-1.17.0.md

```diff
@@ -0,0 +1,481 @@
+that detects potential correctness risks and provides optimization suggestions.
+For example, If a [NDU (Non-deterministic
+Updates)](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/concepts/determinism/#3-determinism-in-streaming-processing)
+issue is detected by EXPLAIN PLAN_ADVICE, the optimizer will append the advice
+at the end of the physical plan, the optimizer will tag the advice id to the
+relational node of related operations, and recommend that users update their
+configurations accordingly. By providing users with this specific advice, the
+optimizer can help them improve the accuracy and reliability of their query
+results. 
+
+
+```
+== Optimized Physical Plan With Advice ==
+...
+
+advice[1]: [WARNING] The column(s): day(generated by non-deterministic function: CURRENT_TIMESTAMP ) can not satisfy the determinism requirement for correctly processing update message('UB'/'UA'/'D' in changelogMode, not 'I' only), this usually happens when input node has no upsertKey(upsertKeys=[{}]) or current node outputs non-deterministic update messages. Please consider removing these non-deterministic columns or making them deterministic by using deterministic functions.
+```
+
+
+The PLAN_ADVICE also helps users improve the performance and efficiency of their
+queries. For example, when a GroupAggregate operation is detected and can be
+optimized to a more efficient local-global aggregation. By providing users with
+this specific advice for optimization, the optimizer enables users to easily
+improve the performance and efficiency of their queries.
+
+
+```
+== Optimized Physical Plan With Advice ==
+...
+advice[1]: [ADVICE] You might want to enable local-global two-phase optimization by configuring ('table.optimizer.agg-phase-strategy' to 'AUTO').
+```
+
+
+In addition, Flink 1.17 resolved several incorrect plan optimizations which led
+to incorrect results reported in
+[FLINK-29849](https://issues.apache.org/jira/browse/FLINK-29849),
+[FLINK-30006](https://issues.apache.org/jira/browse/FLINK-30006), and
+[FLINK-30841](https://issues.apache.org/jira/browse/FLINK-30841).
+
+
+## Checkpoint Improvements 
+
+Generic Incremental Checkpoint (GIC) aims to improve the speed and stability of
+the checkpoint procedure. Some experimental results in the WordCount case are
+shown as below. More details can be found in this [blog
+post](https://www.ververica.com/blog/generic-log-based-incremental-checkpoint). 
+
+Table1: Benefits after enabling GIC in WordCount case
+
+<center>
+<img src="/img/blog/2023-03-09-release-1.17.0/beneifts-after-enabling-gic-in-wordcount-case.png" style="width:90%;margin:15px">
+</center>
+<!-- FIXME: Change to the actual release date -->
+
+Table2: Costs after enabling GIC in WordCount case
+
+<center>
+<img src="/img/blog/2023-03-09-release-1.17.0/costs-after-enabling-gic-in-wordcount-case.png" style="width:90%;margin:15px">
```

Comment:
Not sure whether that's worth the effort, but: The "Full Checkpoint Size" column header is not bold. Why not use the hugo table here instead?

Reply:
😭，Let me try to bold the header

Reply:
done